docs/self-managed/setup/deploy/amazon/openshift/terraform-setup.md

```diff
@@ -0,0 +1,376 @@
+   ```
+
+1. Verify versioning is enabled on the bucket:
+
+   ```bash
+   aws s3api get-bucket-versioning --bucket "$S3_TF_BUCKET_NAME" --region "$AWS_REGION"
+   ```
+
+This S3 bucket will now securely store your Terraform state files with versioning enabled.
+
+#### Create a `config.tf` with the following setup
+
+Once the S3 bucket is created, configure your `config.tf` file to use the S3 backend for managing the Terraform state:
+
+```hcl reference
+https://github.com/camunda/camunda-deployment-references/blob/feature/openshift-ra-standard/aws/rosa-hcp/camunda-versions/8.7/config.tf
+```
+
+#### Initialize Terraform
+
+Once your `config.tf` and authentication are set up, you can initialize your Terraform project. The previous steps configured a dedicated S3 Bucket (`S3_TF_BUCKET_NAME`) to store your state, and the following creates a bucket key that will be used by your configuration.
+
+Configure the backend and download the necessary provider plugins:
+
+```bash
+export S3_TF_BUCKET_KEY="camunda-terraform/terraform.tfstate"
+
+echo "Storing terraform state in s3://$S3_TF_BUCKET_NAME/$S3_TF_BUCKET_KEY"
+
+terraform init -backend-config="bucket=$S3_TF_BUCKET_NAME" -backend-config="key=$S3_TF_BUCKET_KEY"
+```
+
+Terraform will connect to the S3 bucket to manage the state file, ensuring remote and persistent storage.
+
+### OpenShift cluster module setup
+
+This module sets up the foundational configuration for ROSA HCP and Terraform usage.
+
+We will leverage [Terraform modules](https://developer.hashicorp.com/terraform/language/modules), which allow us to abstract resources into reusable components, simplifying infrastructure management.
+
+The [Camunda-provided module](https://github.com/camunda/camunda-tf-rosa) is publicly available and serves as a robust starting point for deploying a Red Hat OpenShift cluster on AWS using a Hosted Control Plane. It is highly recommended to review this module before implementation to understand its structure and capabilities.
+
+Please note that this module is based on the official [ROSA HCP Terraform module documentation](https://docs.openshift.com/rosa/rosa_hcp/terraform/rosa-hcp-creating-a-cluster-quickly-terraform.html). It is presented as an example for running Camunda 8 in ROSA. For advanced use cases or custom setups, we encourage you to use the official module, which includes vendor-supported features.
+
+#### Set up ROSA authentication
+
+To set up a ROSA cluster, certain prerequisites must be configured on your AWS account. Below is an excerpt from the [official ROSA planning prerequisites checklist](https://docs.openshift.com/rosa/rosa_planning/rosa-cloud-expert-prereq-checklist.html):
+
+1. Verify that your AWS account is correctly configured:
+
+   ```bash
+   aws sts get-caller-identity
+   ```
+
+1. Check if the ELB service role exists:
+
+   ```bash
+   aws iam get-role --role-name "AWSServiceRoleForElasticLoadBalancing"
```

Comment:
just a sidenote, strangest requirement ever. Even with the linked documentation it's like why? 😅

Reply:
didn't have to 😄 , was just surprised that even the upstream docs didn't give insights to this weird dependency.